docs/root/api-docs/xds_protocol.rst

```diff
@@ -427,20 +427,21 @@ names becomes empty, that means that the client is no longer interested in any r
 specified type.
 
 For :ref:`Listener <envoy_v3_api_msg_config.listener.v3.Listener>` and :ref:`Cluster <envoy_v3_api_msg_config.cluster.v3.Cluster>` resource
-types, there is also a "wildcard" mode, which is triggered when the initial request on the stream
-for that resource type contains no resource names. In this case, the server should use
-site-specific business logic to determine the full set of resources that the client is interested
-in, typically based on the client's :ref:`node <envoy_v3_api_msg_config.core.v3.Node>` identification. Note
-that once a stream has entered wildcard mode for a given resource type, there is no way to change
-the stream out of wildcard mode; resource names specified in any subsequent request on the stream
-will be ignored.
+types, there is also a "wildcard" subscription, which is triggered when subscribing to the special
+name "*". In this case, the server should use site-specific business logic to determine the full
+set of resources that the client is interested in, typically based on the client's
+:ref:`node <envoy_v3_api_msg_config.core.v3.Node>` identification.
+
+For historical reasons, if the initial request on the stream for a given resource type contains no
+resource names, the server should treat that as a subscription to "*". A client may later
+unsubscribe to "*", or it may later add a subscription to an additional resource name.
```

Comment:
Note that we are going to break any existing setup that names a singleton resource `*`. Seems a bit far fetched, but technically a breaking change. If we instead used something like `xds_internal://*` here as a name (or something unlikely to collide), we might stand a lower chance of collision but don't feel super strong on this one.

Reply:
I think that no matter what name we pick here, collisions are possible.  I'd rather keep the name short and intuitive.